# Теория вероятностей и математическая статистика

### Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

ДЗ сдаем ссылкой на GitHub

Урок 1.

1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.
    
2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [86]:
from math import factorial

Число сочетаний

$$C_{n}^k =\frac{n!}{k!(n - k)!}$$ 

In [89]:
def combinations(n, k):
    # Число сочетаний из  𝑛  элементов по  𝑘  элементов в каждом (в сочетаниях порядок не важен):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Число размещений

$$C_{n}^k =\frac{n!}{(n - k)!}$$ 

In [90]:
def arrangements(n, k):
    # число размещений из  𝑛  элементов по  𝑘  элементов в каждом. При размещениях порядок важен
    return int(factorial(n) / factorial(n - k))

Число перестановок

$$P_{n} = n!$$ 

In [92]:
def permutations(n):
    # Число перестановок из  𝑛  элементов
    return int(factorial(n))

## Задача 1

Из колоды в 52 карты извлекаются случайным образом 4 карты.

a) Найти вероятность того, что все карты – крести.

б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз

- k = 4 < 52
- порядок не важен

это **Сочетание**.

### a) Найти вероятность того, что все карты – крести

1. Количество карт масти *Крести* n= 52/4=13
2. Количество комбинаций из 4 карт масти крести 

$$C_{13}^4 =\frac{n!}{k!(n - k)!}= \frac{13!}{4!(13-4)!}=715, $$ 

$$n=13, k=4$$

***Возможно 715 комбинаций с 4 картами масти крести. Количество карт масти крести = 14.***

In [18]:
print(int(52/4))
C_13_4=combinations(13, 4)
C_13_4

13


715

3. Количество комбинаций из 4 карт из колоды в 52 карты
$$C_{52}^4 =\frac{n!}{k!(n - k)!}= \frac{52!}{4!(52-4)!}=270725, $$ 

$$n=52, k=4$$

***Возможно 270725 комбинаций с 4 картами из колоды 52 карты.***

In [19]:
C_52_4=combinations(52, 4)
C_52_4

270725

4. Вероятнисть того, что все 4 карты крести
$$P = \frac{C_{13}^4}{C_{52}^4}= \frac{715}{270725}=0.00264 $$ 

In [20]:
P=C_13_4/C_52_4
P

0.0026410564225690276

**Ответ**: Вероятнисть того, что все 4 карты крести **P=0,00264**

### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз

- Событие **B**={появится хотя бы один туз}
- 52-4=48 - количество карт без 4 тузов
- Вероятность события B - P(B):

$$P(B) = 1-\frac{C_{48}^4}{C_{52}^4}= 1 - \frac{194580}{270725}=1 - 0.719 = 0.281$$ 

- $C_{48}^4 = 194580$ - Количество комбинаций по 4 карты из колоды в 52 карт из которой убрали 4 туза
- $C_{52}^4 = 270725$ - Количество комбинаций по 4 карты из колоды в 52 карт
- 1 - полная вероятность
- ${C_{48}^4} / {C_{52}^4}= 0.719$  - Вероятность того, что среди карт не окажется ни одного туза


In [26]:
C_48_4=combinations(48, 4)
C_48_4

194580

In [27]:
C_52_4=combinations(52, 4)
C_52_4

270725

In [30]:
P_b = 1 - C_48_4/C_52_4
P_b

0.2812632745405854

**Ответ**: Вероятнисть того, что все 4 карты крести **P=0,281**

## Задача 2

На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

- k = 3 < 10
- порядок важен

Это **РАЗМЕЩЕНИЕ**

### Решение: 
    
- A - открыть с первой попытки
- Общее количество комбинаций $n=C_{10}^3=\frac{10!}{7!}=8*9*10=720$.
- m=1
- $P=\frac{1}{720}=0.0014$

In [84]:
C_10_3=arrangements(10, 3)
print(C_10_3)
P=1/C_10_3
P

720


0.001388888888888889

### Ответ: Вероятность открыть дверь с первой попытки = 0.14%

## Задача 3

В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

### Решение: 
    
- Тянем 1-ю детадь, с вероятностью 9/15=3/5 вытянем окрашенную деталь.
- Тянем 2-ю деталь. Вероятность вытянуть 2-ю окрашенную деталь = 8/14 = 4/7.
- Тянем 3-ю деталь. Вероятность вытянуть 3-ю окрашенную деталь = 7/13.
- Вероятность 3 раза подряд вытянуть окрашенную деталь = (3/5)*(4/7)*(7/13) = 0.18461538461538457 или около 18,5%

In [73]:
(3/5)*(4/7)*(7/13)

0.18461538461538457

### Ответ: Вероятность 3 раза подряд вытянуть окрашенные детали = 18.5%

## Задача 4

В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

### Решение 1: 
    
- Тянем 1-й билет, с вероятностью 2/100=1/50 вытянем выигрышный билет.
- Тянем 2-й билет. Вероятность вытянуть 2-й выигрышный билет = 1/99.
- Вероятность 2 раза подряд вытянуть выигрыш = (1/50)*(1/99) = 1/4950 = 0.0002 или 0.02%

In [98]:
(1/50)*(1/99)

0.00020202020202020205

### Ответ: Вероятность 2 раза подряд вытянуть выигрыш = 0.02%

In [102]:
P = combinations(2, 2)/combinations(100, 2)
P

0.00020202020202020202

## Задача 4

### Решение 2: 

- Cобытия A - из 2-х купленных билетов ни один не выигрышный
- Событие B - 1 приобретенных билет из 2-х окажется выигрышным
- Событие C - 2 приобретенных билета окажутся выигрышными

- Число исходов, благоприятсятвующих событию 
- Общее число исходов $C_{100}^2=4950$

In [54]:
# Общее число исходов -  - это число способов выбрать любые 2 билета из 100 возможных.
C_100_2=combinations(100, 2)
C_100_2

4950

In [45]:
# Событие B - 1 приобретенных билет bp 2 окажется выигрышным

In [56]:
# один из купленных билетов не выигрышный
С_98_1=combinations(98, 1)
С_98_1

98

In [60]:
# один из 2-х билетов выигрышный
С_2_1=combinations(2, 1)
С_2_1

2

In [61]:
# Количество комбинаций, где хотя бы один билет выигрышный
С1 = С_2_1*С_98_1
С1

196

In [ ]:
#  Количество комбинаций, где из 2-х купленных билетов ни один не выигрышный

In [62]:
C_98_2=combinations(98, 2)
C_98_2

4753

In [63]:
# Вероятность остаться без выигрыша
P_A = C_98_2/C_100_2
P_A

0.9602020202020202

In [65]:
# Вероятность того, что 1 из приобретенных билетов окажется выигрышными
P_B = С1/C_100_2
P_B

0.039595959595959594

In [96]:
# Вероятность того, что 2 приобретенных билета окажутся выигрышными
P_C = 1 - P_A - P_B
P_C

0.00020202020202023802

### Ответ:

- Вероятность  того, что 2 приобретенных билета окажутся выигрышными мизерная P(C)=0.02%, при этом вероятность остаться без выигрыша велика - 96.02%.
- Вероятность того хотя бы один из 2-х приобретенных билетов окажется выигрышным P(B)=3.96%.
- Любая лотерея заведома проигрышна для участника.
    